In [1]:
import pandas as pd
import numpy as np
import os
os.chdir(r"C:\Users\DAY\Desktop\DIPLOMADO_ANALISIS_DE_DATOS")

df=pd.read_csv('supermarket_sales.csv')

In [2]:
df.sample(5)

,Invoice ID,Branch,City,Customer type,Gender,Product line,Unit price,Quantity,Tax 5%,Total,Date,Time,Payment,cogs,gross margin percentage,gross income,Rating
450,891-01-7034,B,Mandalay,Normal,Female,Electronic accessories,74.71,6,22.4130,470.6730,1/1/2019,19:07,Cash,448.26,4.761905,22.4130,6.7
59,120-06-4233,C,Naypyitaw,Normal,Male,Electronic accessories,30.61,6,9.1830,192.8430,3/12/2019,20:36,Cash,183.66,4.761905,9.1830,9.3
204,394-43-4238,B,Mandalay,Member,Male,Home and lifestyle,17.77,5,4.4425,93.2925,2/15/2019,12:42,Credit card,88.85,4.761905,4.4425,5.4
79,756-01-7507,C,Naypyitaw,Normal,Female,Health and beauty,20.38,5,5.0950,106.9950,1/22/2019,18:56,Cash,101.90,4.761905,5.0950,6.0
389,596-42-3999,B,Mandalay,Normal,Male,Food and beverages,18.22,7,6.3770,133.9170,3/10/2019,14:04,Credit card,127.54,4.761905,6.3770,6.6


In [3]:
#Transforma las columnas a un array NumPy
unitp_np= df[['Unit price']].to_numpy()
quantity_np= df[['Quantity']].to_numpy()
tax_np= df[['Tax 5%']].to_numpy()
total_np= df[['Total']].to_numpy()
cogs_np= df[['cogs']].to_numpy()
grossmp_np= df[['gross margin percentage']].to_numpy()
grossin_np= df[['gross income']].to_numpy()
rating_np= df[['Rating']].to_numpy()

In [5]:
#Obtener la media de Unit price 
media=unitp_np.mean()
format_media= "{:.2f}".format(media)
print("La media de Unit price es =", format_media)

#Obtener la media de Quantity
media=quantity_np.mean()
format_media= "{:.2f}".format(media)
print("La media de Quantity es =", format_media)

La media de Unit price es = 55.67
La media de Quantity es = 5.51


In [11]:
#Obtener la mediana de Unit price 
mediana=np.median(unitp_np)
print(f"La mediana de Unit price es = {mediana:.2f}")

#Obtener la mediana de Quantity
mediana=np.median(quantity_np)
print(f"La mediana de Quantity es = {mediana:.2f}")

La mediana de Unit price es = 55.23
La mediana de Quantity es = 5.00


In [12]:
#Obtener la moda de Unit price 
vals, counts= np.unique(unitp_np, return_counts=True)
index=np.argmax(counts) 
moda= vals[index]
print(f'La moda de Unit price es = {moda:.2f}')

#Obtener la moda de Quantity
vals, counts= np.unique(quantity_np, return_counts=True)
index=np.argmax(counts) 
moda= vals[index]
print(f'La moda de Quantity es = {moda:.2f}')

La moda de Unit price es = 83.77
La moda de Quantity es = 10.00


### Ticket promedio

In [13]:
#Total por ticket
tickets = (df.groupby("Invoice ID", as_index=False).agg(ticket_total=("Total", "sum"),City=("City", "first"),Gender=("Gender", "first")))
tickets.head()

,Invoice ID,ticket_total,City,Gender
0,101-17-6199,336.5565,Yangon,Male
1,101-81-4070,131.9220,Naypyitaw,Female
2,102-06-2002,132.5625,Naypyitaw,Male
3,102-77-2261,480.0285,Naypyitaw,Male
4,105-10-6182,45.1080,Yangon,Male


In [14]:
#Ticket promedio por ciudad
ticket_promedio_ciudad = (tickets.groupby("City")["ticket_total"].mean())
ticket_promedio_ciudad.round(2)

City
Mandalay     319.87
Naypyitaw    337.10
Yangon       312.35
Name: ticket_total, dtype: float64

In [15]:
#Ticket promedio por product line y ciudad
ticket_pl_ciudad = (df.groupby(["Invoice ID", "City", "Product line"], as_index=False).agg(ticket_total=("Total", "sum")))
ticket_promedio_pl_ciudad = (ticket_pl_ciudad.groupby(["City", "Product line"])["ticket_total"].mean())
ticket_promedio_pl_ciudad.round(2)

City       Product line          
Mandalay   Electronic accessories    310.03
           Fashion accessories       264.73
           Food and beverages        304.30
           Health and beauty         376.99
           Home and lifestyle        350.98
           Sports and travel         322.39
Naypyitaw  Electronic accessories    344.89
           Fashion accessories       331.69
           Food and beverages        360.10
           Health and beauty         319.53
           Home and lifestyle        308.79
           Sports and travel         350.27
Yangon     Electronic accessories    305.29
           Fashion accessories       320.25
           Food and beverages        295.92
           Health and beauty         268.04
           Home and lifestyle        344.88
           Sports and travel         328.35
Name: ticket_total, dtype: float64

In [16]:
#Ticket promedio por género
ticket_promedio_genero = (tickets.groupby("Gender")["ticket_total"].mean())
ticket_promedio_ciudad.round(2)

City
Mandalay     319.87
Naypyitaw    337.10
Yangon       312.35
Name: ticket_total, dtype: float64

In [17]:
#Ticket promedio por genero y product line
ticket_genero_pl = (df.groupby(["Invoice ID", "Gender", "Product line"], as_index=False).agg(ticket_total=("Total", "sum")))
ticket_promedio_genero_pl = (ticket_genero_pl.groupby(["Gender", "Product line"])["ticket_total"].mean())
ticket_promedio_genero_pl.round(2)

Gender  Product line          
Female  Electronic accessories    322.64
        Fashion accessories       317.06
        Food and beverages        368.57
        Health and beauty         290.02
        Home and lifestyle        380.21
        Sports and travel         324.71
Male    Electronic accessories    316.69
        Fashion accessories       291.08
        Food and beverages        273.50
        Health and beauty         348.10
        Home and lifestyle        294.14
        Sports and travel         340.36
Name: ticket_total, dtype: float64

- El ticket promedio se calcula tomando el monto total de cada factura (Invoice ID) y obteniendo el promedio de dichos montos según la categoría solicitada (ciudad, género, línea de producto). Se utiliza la columna Total porque representa el valor final pagado por el cliente, incluyendo impuestos.

### Top 5 ventas

In [18]:
#Por ciudad
ventas_ciudad = df.groupby("City")["Total"].sum()
ventas_ciudad

City
Mandalay     106197.6720
Naypyitaw    110568.7065
Yangon       106200.3705
Name: Total, dtype: float64

In [19]:
ranking_ciudad = ventas_ciudad.rank(ascending=False, method="dense")
ranking_ciudad

City
Mandalay     3.0
Naypyitaw    1.0
Yangon       2.0
Name: Total, dtype: float64

In [20]:
top5_ciudad = (pd.DataFrame({"ventas": ventas_ciudad,"rank": ranking_ciudad})
    .query("rank <= 5").sort_values("ventas", ascending=False))
top5_ciudad

,ventas,rank
City,,
Naypyitaw,110568.7065,1.0
Yangon,106200.3705,2.0
Mandalay,106197.6720,3.0


In [21]:
total_ventas = df["Total"].sum()
top5_ciudad["aporte_%"] = (top5_ciudad["ventas"] / total_ventas) * 100
top5_ciudad.round(2)

,ventas,rank,aporte_%
City,,,
Naypyitaw,110568.71,1.0,34.24
Yangon,106200.37,2.0,32.88
Mandalay,106197.67,3.0,32.88


In [22]:
#Por customer
ventas_member = df.groupby("Customer type")["Total"].sum()
ranking_member = ventas_member.rank(ascending=False, method="dense")

top5_member = (pd.DataFrame({"ventas": ventas_member, "rank": ranking_member})
      .query("rank <= 5").sort_values("ventas", ascending=False))

top5_member["aporte_%"] = (top5_member["ventas"] / df["Total"].sum()) * 100
top5_member.round(2)

,ventas,rank,aporte_%
Customer type,,,
Member,164223.44,1.0,50.85
Normal,158743.30,2.0,49.15


In [23]:
#Por método de pago
ventas_payment = df.groupby("Payment")["Total"].sum()
ranking_payment = ventas_payment.rank(ascending=False, method="dense")

top5_payment = (pd.DataFrame({"ventas": ventas_payment, "rank": ranking_payment})
      .query("rank <= 5").sort_values("ventas", ascending=False))

top5_payment["aporte_%"] = (top5_payment["ventas"] / df["Total"].sum()) * 100
top5_payment.round(2)

,ventas,rank,aporte_%
Payment,,,
Cash,112206.57,1.0,34.74
Ewallet,109993.11,2.0,34.06
Credit card,100767.07,3.0,31.20


### Matriz de correlación

In [24]:
df["Hour"] = pd.to_datetime(df["Time"]).dt.hour
df[["Time", "Hour"]].head()

C:\Users\DAY\AppData\Local\Temp\ipykernel_2236\1763166464.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df["Hour"] = pd.to_datetime(df["Time"]).dt.hour


,Time,Hour
0,13:08,13
1,10:29,10
2,13:23,13
3,20:33,20
4,10:37,10


In [25]:
corr_hora_total = df[["Hour", "Total"]].corr()
corr_hora_total

,Hour,Total
Hour,1.00000,-0.00277
Total,-0.00277,1.00000


- La correlación entre la hora de la transacción y el total de venta es prácticamente nula (-0.0028), lo que indica que no existe una relación lineal significativa entre ambas variables. Esto sugiere que el monto de compra no está influenciado por la hora del día.

In [26]:
corr_precio_rating = df[["Unit price", "Rating"]].corr()
corr_precio_rating

,Unit price,Rating
Unit price,1.000000,-0.008778
Rating,-0.008778,1.000000


- La correlación entre el precio unitario y la calificación del cliente es prácticamente nula (-0.0088), lo que indica que el costo del producto no influye de manera significativa en la percepción del cliente medida a través del rating.